### PR TITLE
fix(react): maintain focus on pagination button when last or first page is reached

### PR DESCRIPTION
**Pull request by thuey**
Closes https://github.com/dequelabs/cauldron/issues/703

Trying to manage focus when child components are replaced seems error-prone. Instead, I got rid of the tabstop component and made the button component behave like the tabstop component when disabled.

### CODE REVIEW

File: docs/Demo/index.js [hunk 1]
```diff
@@ -52,16 +52,18 @@ const Demo = props => {
...
             const componentMarkup = renderState(thinState);
+            const key = DEMO_key || componentMarkup;
             const afterMarkup =
               DEMO_renderAfter &&
               !DEMO_hide_renderAfterBefore &&
               jsxStringify(DEMO_renderAfter, stringifyConfig);
 
             return (
-              <div key={componentMarkup}>
+              <div key={key}>
```
Comment (thuey) [inline docs/Demo/index.js:66]: Focus wasn't being maintained at all when paging, whether you were on the last page or not. That's because the DOM was getting remounted every time the markup changed. I added the ability to provide a key

File: packages/react/__tests__/src/components/Pagination/index.js [hunk 2]
```diff
@@ -28,29 +28,39 @@ describe('Pagination', () => {
           wrapper
             .find('button')
             .at(0)
-            .is('[aria-disabled="true"]')
-        ).toBe(true);
+            .getDOMNode()
+            .getAttribute('aria-disabled')
+        ).toBe('true');
```
Comment (thuey) [inline packages/react/__tests__/src/components/Pagination/index.js:33]: Enzyme struggles when you set `aria-disabled={true | false}` (boolean). Before, the TooltipTabstop was setting `aria-disabled="true"`.

File: packages/react/src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -184,7 +130,8 @@ Pagination.propTypes = {
   onLastPageClick: PropTypes.func,
   // @ts-expect-error
   tooltipPlacement: PropTypes.string,
-  className: PropTypes.string
+  className: PropTypes.string,
+  thin: PropTypes.bool
```
Comment (thuey) [inline packages/react/src/components/Pagination/Pagination.tsx:134]: Not part of this PR, but a linting issue was raised with `thin` not being defined on prop types

**Review by scurker — APPROVED** [review]
It looks like the styles did change slightly, but I don't think that's a flaw with what you did. It looks like the active state has a background color while in dark mode for disabled pagination elements which probably shouldn't be happening.

File: packages/react/__tests__/src/components/Pagination/index.js [hunk 4]
```diff
@@ -28,29 +28,39 @@ describe('Pagination', () => {
...
           wrapper
             .find('button')
             .at(1)
-            .is('[aria-disabled="true"]')
-        ).toBe(true);
+            .getDOMNode()
+            .getAttribute('aria-disabled')
+        ).toBe('true');
 
+        console.log(
```
Comment (stephenmathieson) [inline packages/react/__tests__/src/components/Pagination/index.js:43]: 👀

Comment (thuey) [reply packages/react/__tests__/src/components/Pagination/index.js:43]: Good 👀 . I'll fix it. At least it was just a test file

Comment (thuey) [reply packages/react/__tests__/src/components/Pagination/index.js:43]: https://github.com/dequelabs/cauldron/pull/941
